src/frontend/context/Vilkårsvurdering/vilkårsvurdering.test.ts

```diff
@@ -0,0 +1,362 @@
+        expect(nyVilkårsvurdering[1].periode).toStrictEqual(nyPeriode('2020-05-01', undefined));
+        expect(nyVilkårsvurdering[1].resultat).toBe(Resultat.NEI);
+    });
+
+    test('Skal legge til nytt vilkår med likt resultat som lager to perioder (uten sammenslåing)', () => {
+        const vilkårForPerson = [
+            mockVilkår(VilkårType.BOSATT_I_RIKET, Resultat.JA, nyPeriode('2020-01-01')),
+        ];
+
+        const nyVilkårsvurdering = leggTilNyttVilkårIGjeldendeVilkårsvurdering(
+            vilkårForPerson,
+            mockVilkår(VilkårType.BOSATT_I_RIKET, Resultat.JA, nyPeriode('2020-05-01'))
+        );
+        expect(nyVilkårsvurdering.length).toBe(2);
+        expect(nyVilkårsvurdering[0].periode).toStrictEqual(nyPeriode('2020-01-01', '2020-04-30'));
+        expect(nyVilkårsvurdering[0].resultat).toBe(Resultat.JA);
+        expect(nyVilkårsvurdering[1].periode).toStrictEqual(nyPeriode('2020-05-01', undefined));
+        expect(nyVilkårsvurdering[1].resultat).toBe(Resultat.JA);
+    });
+
+    test('Skal ikke slå sammen to ulike vilkår', () => {
+        const vilkårForPerson = [
+            mockVilkår(
+                VilkårType.BOSATT_I_RIKET,
+                Resultat.JA,
+                nyPeriode('2020-01-01', '2020-03-31')
+            ),
+            mockVilkår(
+                VilkårType.LOVLIG_OPPHOLD,
+                Resultat.JA,
+                nyPeriode('2020-01-01', '2020-06-15')
+            ),
+        ];
+
+        const nyVilkårsvurdering = slåSammenVilkårForPerson(
+            vilkårForPerson,
+            VilkårType.BOSATT_I_RIKET
+        );
+        expect(nyVilkårsvurdering.length).toBe(2);
+    });
+
+    test('Skal slå sammen tre perioder med samme resultat', () => {
+        const vilkårForPerson = [
+            mockVilkår(
+                VilkårType.BOSATT_I_RIKET,
+                Resultat.JA,
+                nyPeriode('2020-01-01', '2020-03-31')
+            ),
+            mockVilkår(
+                VilkårType.BOSATT_I_RIKET,
+                Resultat.JA,
+                nyPeriode('2020-04-01', '2020-06-15')
+            ),
+            mockVilkår(
+                VilkårType.BOSATT_I_RIKET,
+                Resultat.JA,
+                nyPeriode('2020-06-16', '2021-01-01')
+            ),
+            mockVilkår(
+                VilkårType.LOVLIG_OPPHOLD,
+                Resultat.JA,
+                nyPeriode('2020-01-01', '2020-06-15')
+            ),
+        ];
+
+        const nyVilkårsvurdering = slåSammenVilkårForPerson(
+            vilkårForPerson,
+            VilkårType.BOSATT_I_RIKET
+        );
+        expect(nyVilkårsvurdering.length).toBe(2);
+    });
+
+    test('Skal slå sammen første 2 perioder og fylle hull med nytt aksjonspunkt etter det', () => {
+        const vilkårForPerson = [
+            mockVilkår(
+                VilkårType.BOSATT_I_RIKET,
+                Resultat.JA,
+                nyPeriode('2020-01-01', '2020-03-31')
```

Comment:
Bør man teste håndtering av perioder som ikke har perfekte datoer back-to-back, men overlappende? At vi får en forventet feil hvis vi har glemt å flytte datoer osv på periodene i forkant?

Reply:
Ja, det gjør vel den neste testen som fyller hullet? Vet ikke om det bør feile?
Har forsøkt å dele opp i tester som isolerer de ulike funksjonene og tester hele contexten, men kunne sikkert hatt flere tester på hele "verdikjeden".

Reply:
Tenkte ikke på hull, men "ulovlig" overlappende, hvis denne linja feks hadde vært `nyPeriode('2020-01-01', '2020-04-02')` :) men ikke sikkert det er nødvendig å sjekke håndtering av det

Reply:
mhm, usikker på hva systemet eventuelt skal gjøre da. Jeg tenker den ikke trenger å gjøre noe? Hva tenker du @ninakjek ?

Reply:
Evt kanskje gi en error med beskjed om at den har blitt bedt om å evaluere noe den ikke er i stand til?

Reply:
hmm, ja, men litt usikker på hvordan det vil gå ut over brukeren. Tror vi kanskje kan se på det senere. Litt usikker på om det hjelper å "kaste en exception".